backend/lib/deal-observer.js

```diff
@@ -1,11 +1,102 @@
 /** @import {Queryable} from '@filecoin-station/deal-observer-db' */
-/** @import {Provider} from 'ethers' */
+/** @import { BlockEvent } from './rpc-service/data-types.js' */
+import assert from 'node:assert'
+import { getActorEvents, getActorEventsFilter } from './rpc-service/service.js'
+import { ActiveDealDbEntry } from '@filecoin-station/deal-observer-db/lib/types.js'
+import { Value } from '@sinclair/typebox/value'
+import { fromJSON, toJSON } from 'multiformats/cid'
 
 /**
- * Observe the built-in actor events on the Filecoin blockchain to discover new FIL+ deals
+ * @param {number} blockHeight
  * @param {Queryable} pgPool
- * @param {Provider} rpcProvider
+ * @param {(method:string,params:object) => object} makeRpcRequest
+ * @returns {Promise<void>}
  */
-export const observeBuiltinActorEvents = async (pgPool, rpcProvider) => {
-  // TBD
+export async function observeBuiltinActorEvents (blockHeight, pgPool, makeRpcRequest) {
+  const eventType = 'claim'
+  const activeDeals = await getActorEvents(getActorEventsFilter(blockHeight, eventType), makeRpcRequest)
+  assert(activeDeals !== undefined, `No ${eventType} events found in block ${blockHeight}`)
+  console.log(`Observed ${activeDeals.length} ${eventType} events in block ${blockHeight}`)
+  await storeActiveDeals(activeDeals, pgPool)
+}
+
+/**
+ * @param {Queryable} pgPool
+ * @returns {Promise<ActiveDealDbEntry | null>}
+ */
+export async function fetchDealWithHighestActivatedEpoch (pgPool) {
+  const query = 'SELECT * FROM active_deals ORDER BY activated_at_epoch DESC LIMIT 1'
+  const result = (await pgPool.query(query)).rows.map((row) => {
```

Comment:
I would find this to be easier to read, as line 29 is doing a few things with ( ) as well

```suggestion
  const { rows } = await pgPool.query('SELECT * FROM active_deals ORDER BY activated_at_epoch DESC LIMIT 1')
  const result = rows.map((row) => {
```